PGIP/Server.hs

```diff
@@ -230,8 +232,11 @@ hetsServer opts1 = do
            -- only otherwise stick to the old response methods
            else oldWebApi newOpts tempLib permFile sessRef re pathBits qr2
              meth respond
-
+#ifdef WARP1
```

Comment:
you should make a conditional "type ResIO a = IO a" (or "ResourceT IO a") and also use it within "type  WebResponse" (about line 158). This would avoid the #ifdefs for the type signatures.